Library/Homebrew/os/linux/elf.rb

```diff
@@ -172,6 +188,18 @@ def needed_libraries_using_readelf(path)
     end
   end
 
+  def patcher
+    begin
+      require "patchelf"
+    rescue LoadError
+      ## :: PATCHELF_VERSION
+      Homebrew.install_gem! "patchelf", version: '1.0.0'
+      require "patchelf"
+    end
+
+    @patcher ||= PatchELF::Patcher.new to_s, logging: false
```

Comment:
Maybe do `||= begin` for this entire method, so that we don't try to `require` and `rescue` when `@patcher` is already memoized?